PWGEM/PhotonMeson/Tasks/Pi0EtaToGammaGamma.cxx

```diff
@@ -422,28 +472,108 @@ struct Pi0EtaToGammaGamma {
+    if (photons_coll.size() < 3) {
+      return;
+    }
+    const double rotationAngle = M_PI / 2.0; // rotaion angle 90°
+    // ROOT::Math::XYZVector meson3D(meson.Px(), meson.Py(), meson.Pz());
+    ROOT::Math::AxisAngle rotationAxis(meson.Vect(), rotationAngle);
+    // LOG(info) << "rotationAxis.Angle() = " << rotationAxis.Angle();
+    ROOT::Math::Rotation3D rotationMatrix(rotationAxis);
+    // ROOT::Math::XYZVector test(photon1.Px(), photon1.Py(), photon1.Pz());
+    // ROOT::Math::XYZVector photon1_3D(photon1.Px(), photon1.Py(), photon1.Pz());
+    // ROOT::Math::XYZVector photon2_3D(photon2.Px(), photon2.Py(), photon2.Pz());
+
+    // float openingAngleTest = std::acos(photon1_3D.Dot(test) / (std::sqrt(photon1_3D.Mag2()) * std::sqrt(test.Mag2())));
+    // LOG(info) << "openingAngleTest before rotation = " << openingAngleTest;
+    photon1 = rotationMatrix * photon1;
+    photon2 = rotationMatrix * photon2;
+
+    // openingAngleTest = std::acos(photon1_3D.Dot(test) / (std::sqrt(photon1_3D.Mag2()) * std::sqrt(test.Mag2())));
+    // LOG(info) << "openingAngleTest = " << openingAngleTest;
+
+    for (auto& photon : photons_coll) {
+      if (photon.globalIndex() == ig1 || photon.globalIndex() == ig2) {
+        // only combine rotated photons with other photons
+        continue;
+      }
+      ROOT::Math::PtEtaPhiMVector photon3;
+      photon3.SetPt(photon.pt(0.));
+      photon3.SetEta(photon.eta());
+      photon3.SetPhi(photon.phi());
+      photon3.SetM(0.);
+      // ROOT::Math::XYZVector photon3_3D(photon3.Px(), photon3.Py(), photon3.Pz());
+      ROOT::Math::PtEtaPhiMVector mother1 = photon1 + photon3;
+      ROOT::Math::PtEtaPhiMVector mother2 = photon2 + photon3;
+
+      float openingAngle1 = std::acos(photon1.Vect().Dot(photon3.Vect()) / (photon1.P() * photon3.P()));
+      float openingAngle2 = std::acos(photon2.Vect().Dot(photon3.Vect()) / (photon2.P() * photon3.P()));
+      // float openingAngle1_1 = std::acos(photon1_3D.Dot(photon3_3D) / (std::sqrt(photon1_3D.Mag2()) * std::sqrt(photon3_3D.Mag2())));
+      // float openingAngle2_2 = std::acos(photon2_3D.Dot(photon3_3D) / (std::sqrt(photon2_3D.Mag2()) * std::sqrt(photon3_3D.Mag2())));
+      // LOG(info) << "openingAngle1 = " << openingAngle1;
+      // LOG(info) << "openingAngle2 = " << openingAngle2;
+      // LOG(info) << "openingAngle1_1 = " << openingAngle1_1;
+      // LOG(info) << "openingAngle2_2 = " << openingAngle2_2;
+
+      // Fill histograms
+      if (openingAngle1 > minOpenAngle) {
+        registry.fill(HIST("EMCEMC/h2MggPt_Rotated"), mother1.M(), mother1.Pt());
+      }
+      if (openingAngle2 > minOpenAngle) {
+        registry.fill(HIST("EMCEMC/h2MggPt_Rotated"), mother2.M(), mother2.Pt());
+      }
+    }
+  }
+
+  // Filter collisionFilter_PCM_mix = nabs(o2::aod::collision::posZ) < 10.f && o2::aod::collision::numContrib > (uint16_t)0 && o2::aod::evsel::sel8 == true && o2::aod::emreducedevent::ngpcm > 0;
```

Comment:
You can keep Filter like this and use it for all sub systems.
Filter collisionFilter_PCM_mix = nabs(o2::aod::collision::posZ) < 10.f && o2::aod::collision::numContrib > (uint16_t)0 && o2::aod::evsel::sel8 == true;

Reply:
Yeah good point. But also for the same event than, right?

Reply:
No. In the same event, the number of events are counted in a histogram.